lib/rubocop/cop/rspec/described_class.rb

```diff
@@ -156,64 +156,69 @@ def offensive_described_class?(node)
 
           return if nearest_described_class.equal?(node)
 
-          full_const_name(nearest_described_class) == full_const_name(node)
+          FullConstNameFinder.call(nearest_described_class) ==
+            FullConstNameFinder.call(node)
         end
 
-        def full_const_name(node)
-          collapse_namespace(namespace(node), const_name(node))
-        end
+        module FullConstNameFinder
+          module_function
 
-        # @param namespace [Array<Symbol>]
-        # @param const [Array<Symbol>]
-        # @return [Array<Symbol>]
-        # @example
-        #   # nil represents base constant
-        #   collapse_namespace([], :C)                 # => [:C]
-        #   collapse_namespace([:A, :B], [:C)          # => [:A, :B, :C]
-        #   collapse_namespace([:A, :B], [:B, :C)      # => [:A, :B, :C]
-        #   collapse_namespace([:A, :B], [nil, :C)     # => [nil, :C]
-        #   collapse_namespace([:A, :B], [nil, :B, :C) # => [nil, :B, :C]
-        def collapse_namespace(namespace, const)
-          return const if namespace.empty?
-          return const if const.first.nil?
-
-          start = [0, (namespace.length - const.length)].max
-          max = namespace.length
-          intersection = (start..max).find do |shift|
-            namespace[shift, max - shift] == const[0, max - shift]
+          def call(node)
```

Comment:
If not the docs, I would struggle to understand what those methods do.
Do you mind to add docs for `call` as well?

Reply:
Done